From 7ba71967a2678e119bcd819ac92d08bd038e1879 Mon Sep 17 00:00:00 2001
From: Myy <myy@miouyouyou.fr>
Date: Mon, 17 Jul 2017 18:34:33 +0000
Subject: [PATCH 5/5] Mali midgard r19p0 fixes for 4.13 kernels

Mostly due to DMA structure members renamed, and some signatures
modifications done during the development of Linux 4.11.

* mm, fs: reduce fault, page_mkwrite, and pfn_mkwrite to take only vmf
  11bac80004499ea59f361ef2a5516c84b6eab675
  v4.11-rc1

* sched/headers: Move task->mm handling methods to <linux/sched/mm.h>
  68e21be2916b359fd8afb536c1911dc014cfd03e
  v4.11-rc1

* sched/headers: Move task-stack related APIs from <linux/sched.h> to
  <linux/sched/task_stack.h>
  f3ac60671954c8d413532627b1be13a76f394c49
  v4.11-rc1

* dma-fence: Introduce drm_fence_set_error() helper
  a009e975da5c7d42a7f5eaadc54946eb5f76c9af
  v4.11-rc1

Signed-off-by: Myy <myy@miouyouyou.fr>

Patch adapted from the one at https://github.com/armbian/build/tree/master/patch/kernel/rockchip-next, revision c55f6ea8bc220bded6695fdae8b0c5bfa891b688

Upstream-Status: Inappropriate [not author]
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 drivers/gpu/arm/midgard/mali_kbase.h           | 2 ++
 drivers/gpu/arm/midgard/mali_kbase_mem_linux.c | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/arm/midgard/mali_kbase.h b/drivers/gpu/arm/midgard/mali_kbase.h
index 443d4b1..28a8165 100755
--- a/drivers/gpu/arm/midgard/mali_kbase.h
+++ b/drivers/gpu/arm/midgard/mali_kbase.h
@@ -35,6 +35,8 @@
 #include <linux/mutex.h>
 #include <linux/rwsem.h>
 #include <linux/sched.h>
+#include <linux/sched/task_stack.h> 
+#include <linux/sched/mm.h>
 #include <linux/slab.h>
 #include <linux/spinlock.h>
 #include <linux/vmalloc.h>
diff --git a/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c b/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c
index b6dac55..c72d654 100755
--- a/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c
+++ b/drivers/gpu/arm/midgard/mali_kbase_mem_linux.c
@@ -1885,8 +1885,9 @@ static void kbase_cpu_vm_close(struct vm_area_struct *vma)
 KBASE_EXPORT_TEST_API(kbase_cpu_vm_close);
 
 
-static int kbase_cpu_vm_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
+static int kbase_cpu_vm_fault(struct vm_fault *vmf)
 {
+	struct vm_area_struct *vma = vmf->vma;
 	struct kbase_cpu_mapping *map = vma->vm_private_data;
 	pgoff_t rel_pgoff;
 	size_t i;
-- 
2.7.4

